src/cudamatrix/cu-math.cc

```diff
@@ -233,6 +259,15 @@ void Randomize(const CuMatrixBase<double> &src,
                const CuArray<int32> &copy_from_idx,
                CuMatrixBase<double> *tgt);
 
+template
```

Comment:
Since this is kind of a special purpose function, you don't have to instantiate for both float and double-- you can just hard-code it to BaseFloat.